cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/compute/GcpInstanceResourceBuilder.java

```diff
@@ -213,10 +217,15 @@ String getPublicKey(String groupPublicKey, String groupLoginUserName) {
         return publicKey;
     }
 
-    private String getHostname(CloudStack cloudStack, List<CloudResource> buildableResource) {
+    private String getHostname(CloudStack cloudStack, int privateId) {
         String hostname = null;
-        if (!cloudStack.getGroups().isEmpty() && !cloudStack.getGroups().get(0).getInstances().isEmpty()) {
-            hostname = cloudStack.getGroups().get(0).getInstances().get(0).getStringParameter(CloudInstance.DISCOVERY_NAME);
+        if (!cloudStack.getGroups().isEmpty() && !cloudStack.getGroups().get(0).getInstances().isEmpty()
+                && cloudStack.getGroups().get(0).getInstances().get(privateId) != null) {
+            hostname = cloudStack.getGroups().get(0).getInstances().get(privateId).getStringParameter(CloudInstance.DISCOVERY_NAME);
+            // Setting hostnames for only FreeIPA hosts
+            if (hostname == null || !hostname.startsWith("ipaserver")) {
```

Comment:
I think we should implement a cloudStack.type parameter which can get FREEIPA, STACK  values

Reply:
This is a great suggestion. Could you please help me with how the stack name for freeipa got FreeIPA in it? Looks like that is the last place the information is dealt with as an Enum, beyond that they are blended as a string in names. I tried to track that yesterday, but got lost and implemented this simple solution.

Reply:
Hi @doktoric, I tried what seems simple and gets the job done with respect to stack type parameter suggestion. Please review.